### ISSUE TITLE
`nix-instantiate` does not report failures about files copied to `/nix/store/*`

### ISSUE THREAD

**Issue by starptr** [NixOS/nix#7118]
**Describe the bug**

When calling `nix-instantiate` on a derivation that references local files, the intended behavior is for it to copy the local files to `/nix/store/*` to be used in `nix-store -r`. In certain situations the file copy fails, but `nix-instantiate` does not report this and so the error is detected when calling `nix-store -r`.

**Steps To Reproduce**

1. Clone [this simple repro](https://github.com/starptr/repro-nix-deriv-sh-extension/tree/06c1cb7882398a12557b3b713cc53f3373617802)
2. Call `nix-instantiate simple.nix`
3. Then call `nix-store -r <derivation path>`, where `<derivation path>` is reported by `nix-instantiate`
4. See error about missing `rcfile` or `runner`, both of which are used in `simple.nix`

**Expected behavior**

`nix-instantiate` should report that the copy failed, and the reason why the copy failed.

**`nix-env --version` output**

```
nix-env (Nix) 2.11.1
```

**Additional context**

To understand [this related behavior](https://discourse.nixos.org/t/nix-build-arbitrarily-fails-depending-on-names-of-files-used-inside-derivation/22122), `nix-instantiate` needs to be more helpful.

**Comment by starptr** [comment]
Yes you are correct, thanks! I expected `builder.sh` to fail because I was trying to `mv` out from the nix store, but instead the script was able to move them out of the store, making debugging this issue very difficult.

**Comment by thufschmitt** [comment]
> I am using nix-user-chroot, which uses user namespaces on my system. Not sure why that would allow me to modify /nix/store which should be read-only, but that is the observed behavior.

I suspect that's due to `nix-user-chroot` not playing well with the build sandbox (iirc there's some kind of Linux namespaces that can't be nested, and since `nix-user-chroot` is already using these the Nix sandbox can't, hence fewer purity guaranties)

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
